dnst: document new fields (#256)
ooni/spec: data-formats/df-002-dnst.md

- `failure` (`string`; nullable): if there was an error, this field is
a string indicating the error, otherwise it MUST be `null`.

- `getaddrinfo_error` (`int`; optional; since 2022-08-23): if the
underlying engine used `getaddrinfo` and `getaddrinfo` failed, this
is the system-dependent error code returned by `getaddrinfo`;

- `hostname` (`string`): the hostname used in the DNS query, which MUST
be reversed for PTR lookups like `1.0.0.127.in-addr.arpa.`.

- `query_type`: (`string`): a valid DNS query type (e.g. `MX`)&mdash;see the
note below regarding representing the system/go resolver results;

- `raw_response`: an optional base64 string containing the bytes of the
response for the engines that allow us to observe it;

- `rcode`: an optional int64 containing the DNS rcode inside the `raw_response`

> fro the engines that allow us to observe it;

Author: ```suggestion
for the engines that allow us to observe it;
```

and [ooni/probe-cli#766](https://github.com/ooni/probe-cli/pull/766) for
more details about how we choose the resolver name.

### Representing system/go resolver results

As of 2022-08-23, resolutions performed by the system/go resolver are
represented in two distinct ways:

> 1. all mainline experiments artifically split a lookup into two entries, one

Author: ```suggestion
1. all mainline experiments artifically split a lookup into two queries and replies, one
```

for `A` and the other for `AAAA`;

2. new step-by-step code (which eventually will become the default) represents a

> system/go lookup as a single `ANY` query containing all the `A`, `AAAA`, and

Author: ```suggestion
system/go lookup as a single `ANY` query/reply containing all the `A`, `AAAA`, and
```
